### PR TITLE
De-duplicate prediction results with the history results

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

De-duplicate prediction results with the history results.
A prediction result could be the same as the one found in history. This happens sometimes to the Az predictor.
When used in list view, it's confusing for the user to see 2 same results in the list, both from history and a predictor. So, we de-duplicate the prediction results against the history results.

For now, we only remove the prediction results that are exactly the same (case-sensitive or case-insensitive decided by the user with the 'HistorySearchCaseSensitive' setting, which is `false` by default) as one of the history results.

## PR Checklist

- [x] PR has a meaningful title
    - Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [x] Make sure you've added one or more new tests
- [x] Make sure you've tested these changes in terminals that PowerShell is commonly used in (i.e. conhost.exe, Windows Terminal, Visual Studio Code Integrated Terminal, etc.)
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] Documentation needed at [PowerShell-Docs](https://github.com/MicrosoftDocs/PowerShell-Docs)
        - [ ] Doc Issue filed: 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/PowerShell/PSReadLine/pull/3543)

### CODE REVIEW

**Review by StevenBucher98 — APPROVED** [review]
LGTM, awesome work Dongbo!

Verified previous duplications with Az Predictor are no longer there on this branch

File: PSReadLine/Prediction.Views.cs [hunk 1]
```diff
@@ -435,19 +447,46 @@ private void AggregateSuggestions()
...
+                                    // those skipped ones. This is due to the limitation of the 'OnSuggestionDisplayed' interface method, which didn't
+                                    // assume any prediction results from a predictor could be filtered out at the initial design time. We will have to
+                                    // change the predictor interface to pass in accurate information, such as:
+                                    //   void OnSuggestionDisplayed(Guid predictorId, uint session, int countOrIndex, int[] skippedIndices)
+                                    //
+                                    // However, an interface change has huge impacts. At least, a newer version of PSReadLine will stop working on the
+                                    // existing PowerShell 7+ versions. For this particular issue, the chance that it could happen is low and the impact
+                                    // of the inaccurate feedback is also low, so we should delay this interface change until another highly-demanded
+                                    // change to the interface is required in future (e.g. changes related to supporting OpenAI models).
```
Comment (andyleejordan) [inline PSReadLine/Prediction.Views.cs:488]: Thanks for the detailed note 👍

**Review by andyleejordan — APPROVED** [review]
Looks great, thanks Dongbo!

**Comment by daxian-dbw** [comment]
Thank you both for the reivew!

**Pull request merged**